File: go/01/01.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	count := 0
	var lines []int 
	var line string
	for {
		fmt.Scanf("%s", &line)
		fmt.Println(line)
		if line == "0" {
			break
		}
		lineInt, _ := strconv.Atoi(line)
		lines = append(lines, lineInt)
	}
	
	for i, el := range lines {
		if i == len(lines)-3{
			break
		}
		window1 := el + lines[i+1] + lines[i+2]
		window2 := lines[i+1] + lines[i+2] + lines[i+3]
		if window1 < window2 {
			count++
		}
	}
	fmt.Println(count)
}

File: python/01/01.py
import sys


depths = sys.stdin.read().split('\n')
depths = [int(el) for el in depths]
count = 0

for i in range(len(depths)-3):
	window1 = sum(depths[i:i+3])
	window2 = sum(depths[i+1:i+4])
	print(window1, window2)
	if (window1 < window2):
		count += 1

print(count)
